flake.nix

```diff
@@ -0,0 +1,34 @@
+{
+  description = "Rustlings devShell.";
+
+  inputs = {
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  };
+
+  outputs = { self, flake-compat, flake-utils, nixpkgs }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
```

Comment:
Yeah, I also see the latter pattern with `legacyPackages` used more often, but think the former is cleaner. However, if you're strongly against keeping this as is, I can change it.

Reply:
@winterqt what do you say?

Reply:
It's bad practice to use it, see [here](https://zimbatm.com/notes/1000-instances-of-nixpkgs) (tl;dr: evaluations can be cached across flakes).

I don't think it *really* matters here since we don't depend on any flakes that we pass Nixpkgs to, but other flakes *may* depend on us for whatever reason, so maybe it'd be good to just go with this? Not sure, what do you think?

Reply:
Oh I see, in this case I will change this just in case. I always thought is was a stylist choice...